src/tracks/playabletrack/wavetrack/ui/WaveTrackView.cpp

```diff
@@ -824,6 +826,24 @@ auto WaveTrackSubView::GetMenuItems(
       pClip = pTrack->GetClipAtTime( time );
    }
 
+   auto renameClip = [pTrack, pClip]( const CommandContext &context ){
+      if (pTrack) {
+         auto &view = TrackView::Get(*pTrack);
+         auto pAffordance = view.GetAffordanceControls();
+         if ( auto affordanceControl =
+            std::dynamic_pointer_cast<WaveTrackAffordanceControls>(
+               pAffordance ) ) {
+            // Begin editing of the label, either with a dialog or directly
+            // in the track panel.
+            auto &project = context.project;
+            affordanceControl->StartEditSelectedClipName( project,
```

Comment:
Suggest adding `WaveTrackAffordanceControls::StartEditClipNameAt(AudacityProject&, double time)`  instead, or use events

Reply:
I don't understand the suggestion to use events.

The new test argument makes the function very general, but maybe you think it is too general.  If you don't, you might object that the generalized function should at least be renamed, and I would agree.

Reply:
It's not about generality. Yes, function name now became at bit controversial.

Reply:
Also, right channel does not always have affordance cell, may be in that case call should be redirected to the view of the leader track?

Reply:
> Also, right channel does not always have affordance cell, may be in that case call should be redirected to the view of the leader track?

That is a good point.

Reply:
If events are used, which object would emit the event and which would react to it?